NeoMathEngine/include/NeoMathEngine/ThreadPool.h

```diff
@@ -90,13 +90,20 @@ inline bool GetTaskIndexAndCount( int threadCount, int threadIndex, int fullCoun
 		index = 0;
 		count = fullCount;
 	}
-
 	return count != 0;
 }
 
 inline bool GetTaskIndexAndCount( int threadCount, int threadIndex, int fullCount, int& index, int& count )
 {
-	return GetTaskIndexAndCount(threadCount, threadIndex, fullCount, 1, index, count);
+	return GetTaskIndexAndCount( threadCount, threadIndex, fullCount, 1, index, count );
+}
+
+//------------------------------------------------------------------------------------------------------------
+
+inline bool IsThreadTaskRelevant( int64_t operationCount )
```

Comment:
Let's move this to KMeans. I hope this method won't be used anywhere else in future (I don't recall using IMathEngine anywhere else in TraditionalML)

Reply:
I planned to use it everywhere to decide use the threadPool or not. For example in Function.cpp in #879

Reply:
And it's wrong.

Old code:
`IsOmpRelevant( vectorCount ) -> IsOmpRelevant( vectorCount, MinOmpOperationCount ) -> ( vectorCount > 1 && MinOmpOperationCount >= MinOmpOperationCount ) -> ( vectorCount > 1 )`
which means "use pool if multiple vectors"
Your code:
`IsThreadTaskRelevant( args.Size() ) -> ( matrixHeight >= 32k )`
which means "use pool only when we have at least 32k vectors".

The second parameter of `IsOmpRelevant` isn't used outside of MathEngine. MathEngine was used only in KMeans. That's why I still think we can leave this code in KMeans. We don't have task now to fine-tune and speed-up clusterization. We have more crucial things to speed-up...

Reply:
I agree, will fix it